test/spec.yml

```diff
@@ -1373,3 +1374,346 @@
+            executionModel: concurrent
+            runtime: node8
+          test2triggersfunc2:
+            file: function.js
+            entrypoint: handler
+            executionModel: concurrent
+            runtime: node8
+        EOF
+    - in: bn deploy test2triggersfunc2
+      out: |-
+        Deployed function test2triggersfunc2
+        Invoke with one of:
+          "bn invoke test2triggersfunc2"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: bn trigger upsert test2triggers --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func test2triggersfunc
+      out: CREATED
+    - in: bn trigger upsert test2triggers2 --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func test2triggersfunc2
+      out: CREATED
+
+- test: Test new trigger with bad source uri (bad-path)
+  cleanup:
+    - bn remove testbadtriggerfunc
+  steps:
+    - in: bn create node8 testbadtriggerfunc
+      out: |-
+          Created function * in /home/dockeruser/test
+            (use "bn deploy testbadtriggerfunc" to deploy the function)
+    - in: bn deploy testbadtriggerfunc
+      out: |-
+        Deployed function testbadtriggerfunc
+        Invoke with one of:
+          "bn invoke testbadtriggerfunc"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: bn trigger upsert testbadtrigger --source-uri redis://not-a-valid-redis-uri --target-func testbadtriggerfunc
+      exit: 1
+      err: |-
+        RequestId: *
+        Internal Binaris server error*
+
+- test: Test update trigger with bad source uri (bad-path)
+  cleanup:
+    - bn trigger remove testtriggerbadsourceuri
+    - bn remove testtriggerbadsourceurifunc
+  steps:
+    - in: bn create node8 testtriggerbadsourceurifunc
+      out: |-
+          Created function * in /home/dockeruser/test
+            (use "bn deploy testtriggerbadsourceurifunc" to deploy the function)
+    - in: bn deploy testtriggerbadsourceurifunc
+      out: |-
+        Deployed function testtriggerbadsourceurifunc
+        Invoke with one of:
+          "bn invoke testtriggerbadsourceurifunc"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: bn trigger upsert testtriggerbadsourceuri --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func testtriggerbadsourceurifunc
+      out: CREATED
+    - in: bn trigger upsert testtriggerbadsourceuri --source-uri redis://not-a-valid-redis-uri --target-func testtriggerbadsourceurifunc
+      exit: 1
+      err: |-
+        RequestId: *
+        Internal Binaris server error*
+
+- test: Test new trigger with missing target function (bad-path)
+  steps:
+    - in: bn trigger upsert notexistingfunctrigger --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func nonexistingfunc
+      exit: 1
+      err: |-
+        RequestId: *
+        Error: No such function
+
+- test: Test update trigger with missing target function (bad-path)
```

Comment:
Redundant test.

Reply:
Why?

Reply:
Already tested for create, it's just protocol validation which is the same for update or create.